packages/discord.js/src/client/BaseClient.js

```diff
@@ -18,17 +18,19 @@ class BaseClient extends EventEmitter {
       throw new DiscordjsTypeError(ErrorCodes.InvalidType, 'options', 'object', true);
     }
 
-    if (options.rest?.userAgentAppendix) {
-      // Merging the default options when a custom user agent appendix is supplied
-      // Replaces the discord.js string. Enforce it.
-      options.rest.userAgentAppendix = `${Options.userAgentAppendix} ${options.rest.userAgentAppendix}`;
-    }
-
     /**
      * The options the client was instantiated with
      * @type {ClientOptions}
      */
-    this.options = mergeDefault(Options.createDefault(), options);
+    this.options = mergeDefault(Options.createDefault(), {
+      ...options,
+      rest: {
+        ...options.rest,
+        userAgentAppendix: options.rest?.userAgentAppendix
+          ? `${Options.userAgentAppendix} ${options.rest.userAgentAppendix}`
+          : undefined,
```

Comment:
Shouldn't this default to `Options.userAgentAppendix` if its not provided by users?

Reply:
That's the whole point of `mergeDefault()`.